Core/Core/Inbox/InboxCoursePicker/View/InboxCoursePickerView.swift

```diff
@@ -0,0 +1,202 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public struct InboxCoursePickerView: View {
+    @ObservedObject private var viewModel: InboxCoursePickerViewModel
+
+    init(viewModel: InboxCoursePickerViewModel) {
+        self.viewModel = viewModel
+    }
+
+    public var body: some View {
+        content
+            .navigationTitleStyled(Text("Select Course", bundle: .core).font(.semibold17).foregroundColor(.textDarkest))
+            .navigationBarTitleDisplayMode(.inline)
+    }
+
+    @ViewBuilder
+    private var content: some View {
+        ScrollView {
+            switch viewModel.state {
+            case .loading:
+                ProgressView()
+                    .progressViewStyle(.indeterminateCircle())
+                    .padding(12)
+            case .data:
+                VStack(spacing: 0) {
+                    courses(courses: viewModel.courses)
+                    groups(groups: viewModel.groups)
+                }
+            case .empty, .error:
+                Text("Some error occured", bundle: .core)
+                    .font(.regular17)
+                    .foregroundColor(.textDarkest)
+            }
+        }
+        .frame(maxWidth: .infinity)
+    }
+
+    private var separator: some View {
+        Color.borderMedium
+            .frame(height: 0.5)
+    }
+
+    private func courses(courses: [Course]) -> some View {
+        VStack(spacing: 0) {
+            if !courses.isEmpty {
+                Section(header:
+                        VStack(spacing: 0) {
+                    separator
+                            Text("Courses")
+                                .font(.regular14)
+                                .frame(maxWidth: .infinity, alignment: .leading)
+                                .foregroundStyle(Color.textDark)
+                                .padding(.horizontal, 12)
+                                .padding(.vertical, 16)
+                                .background(Color.backgroundLight)
+                                .accessibilityHeading(.h1)
+                    separator
```

Comment:
Indentation seems off. We could extract maybe the header builder to make this part smaller just like you did with the courseRow method.